src/Sql/Sql/Auditing/Cmdlet/SetAzSqlServerAuditPolicy.cs

```diff
@@ -0,0 +1,192 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.Auditing.Model;
+using Microsoft.Azure.Commands.Sql.Common;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Auditing.Cmdlet
+{
+    [Cmdlet(
+        VerbsCommon.Set,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + DefinitionsCommon.ServerAuditPolicyCmdletsSuffix,
+        DefaultParameterSetName = DefinitionsCommon.ServerParameterSetName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(bool))]
+    public class SetAzSqlServerAuditPolicy : SqlServerAuditPolicyCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.AuditActionGroupsHelpMessage)]
+        public AuditActionGroups[] AuditActionGroup { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.PredicateExpressionHelpMessage)]
+        [ValidateNotNull]
+        public string PredicateExpression { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.BlobStorageTargetState)]
+        [ValidateSet(SecurityConstants.Enabled, SecurityConstants.Disabled, IgnoreCase = false)]
+        [ValidateNotNullOrEmpty]
+        public string BlobStorageTargetState { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.AuditStorageAccountNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.AuditStorageAccountSubscriptionIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public Guid StorageAccountSubscriptionId { get; set; }
```

Comment:
I realized that the pattern you are using is StorageAccountSubscriptionId + StorageAccountName for identifying the storage accounts in these cmdlets. However, our standard identification in these cases is using only `StorageAccountResourceId`. Users are more likely to have the resource ID rather than the name + subscription GUID separated. Also, this would keep things consistent. Please update this everywhere applicable.